primary/src/part_sync_proposer.rs

```diff
@@ -0,0 +1,234 @@
+// Copyright(C) Facebook, Inc. and its affiliates.
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+use config::{SharedCommittee, WorkerId};
+use crypto::{traits::VerifyingKey, Digest, Hash as _, SignatureService};
+use std::cmp::Ordering;
+use tokio::{
+    sync::mpsc::{Receiver, Sender},
+    time::{sleep, Duration, Instant},
+};
+#[cfg(feature = "benchmark")]
+use tracing::info;
+use tracing::{debug, warn};
+use types::{BatchDigest, Certificate, Header, Round};
+
+#[cfg(test)]
+#[path = "tests/part_sync_proposer_tests.rs"]
+pub mod part_sync_proposer_tests;
+
+/// The proposer creates new headers and send them to the core for broadcasting and further processing.
+pub struct PartiallySyncProposer<PublicKey: VerifyingKey> {
```

Comment:
do we really need this separation? the difference of async_proposer and partially_sync_proposer is an extra check per round

Reply:
let me see if we can avoid it (it would be nice)

Reply:
it seems we may, trying it in another PR. In the meantime I make an issue